drivers/syslog/syslog_device.c

```diff
@@ -169,15 +150,7 @@ static inline int syslog_dev_takesem(FAR struct syslog_dev_s *syslog_dev)
 
 static inline void syslog_dev_givesem(FAR struct syslog_dev_s *syslog_dev)
 {
-#ifdef CONFIG_DEBUG_ASSERTIONS
-  pid_t me = getpid();
-  DEBUGASSERT(syslog_dev->sl_holder == me);
```

Comment:
Do we need to keep debug assertion here?

Reply:
it's better to add to mutex.h

Reply:
The other option was to use `DEBUGVERIFY(nxrmutex_unlock(&syslog_dev->sl_lock));`